Dockerfile

```
@@ -0,0 +1,10 @@
+FROM docker.io/library/alpine:latest
+ARG TARGETARCH
+
+RUN apk add --no-cache tini
+
+ADD ./bin/k0sctl-linux-$TARGETARCH /usr/local/bin/k0sctl
+
+ENTRYPOINT ["/sbin/tini", "--"]
+
+CMD ["k0sctl"]
```

k0sctl should work with `scratch`, just need root certs.

```suggestion
FROM golang:1.24-alpine AS builder

RUN apk add --no-cache ca-certificates

WORKDIR /src

COPY go.mod go.sum ./
RUN go mod download

COPY . .

ARG ENVIRONMENT
ARG GIT_COMMIT
ARG TAG_NAME

ENV CGO_ENABLED=0

RUN go build -trimpath -a \
    -tags "netgo,osusergo,static_build" \
    -installsuffix netgo \
    -ldflags "-s -w \
      -X github.com/k0sproject/k0sctl/version.Environment=${ENVIRONMENT} \
      -X github.com/carlmjohnson/versioninfo.Revision=${GIT_COMMIT} \
      -X github.com/carlmjohnson/versioninfo.Version=${TAG_NAME} \
      -extldflags '-static'" \
    -o /out/k0sctl .

FROM scratch

COPY --from=builder /out/k0sctl /k0sctl

COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt

ENTRYPOINT ["/k0sctl"]
CMD ["version"]
```

---

`openSSH` hosts won't work though since it needs the openssh client binary.

(something like:

```yaml
spec:
  hosts:
    openSSH: 
      address: 10.0.0.1
```
)

---

So this is my personal preference, but I like to use a static image instead of `scratch`. I did update the PR to use the multi-build process and to also bakein ssh so that way there is not a feature missing from the container.

https://labs.iximiuz.com/tutorials/pitfalls-of-from-scratch-images